doc/tutorials/ostro-oic-tutorial.md

```diff
@@ -0,0 +1,1030 @@
+# Overview
+
+In this tutorial, we'll show how to develop a simple application using
+Soletta. Although simple, this application will show some important
+features of Soletta, like OIC and HTTP communication, besides GPIO
+interaction.
+Initially, we are going to run the application on desktop [1], but
+later we're going to show how to run it on devices, such as [MinnowBoard
+MAX](http://wiki.minnowboard.org/MinnowBoard_Wiki_Home).
+
+## Step 0
+
+Soletta must be installed on developer's computer. One can use a package
+provide for his or her distro, or [build Soletta](https://github.com/solettaproject/soletta/wiki/How-to-start).
+
+## Step 1
+
+### Flow Based Programming overview
+
+Soletta provides two API [3] for developers. A C API and a flow based
+API. In this tutorial, we're going to show the flow based API.
+The flow based API uses [Flow Based Programming](https://en.wikipedia.org/wiki/Flow-based_programming)
+concepts. It is a higher level API than its C counterpart, and it's
+much easier to use when expressing relationships among application
+components.
+Basically, a flow based application contains 'nodes' which represents
+entities - like sensors or actuators, as well as operations like
+concatenating string. These nodes have input or output 'ports' that
+send or receive 'packets' of information. These ports can be connected
+to express the flow of information among components. For instance:
+
+```
+button(gpio/reader:pin=1) OUT -> IN light(gpio/writer:pin=2)
+```
+
+Here, `button` is a node of node type [4] `gpio/reader` and light is
+another node, of node type `gpio/writer`. A `gpio/reader` node
+has an output port `OUT` - which sends boolean packets - that
+is connected to input port `IN` of the `gpio/writer` - which
+in turn, receives boolean packets as well. So, in this sample,
+we use `button` output to feed `light`. The `pin=N` on those
+nodes is a 'node option'. Node options are used to setup nodes,
+so each node, in this sample, is aware of which GPIO pin they
+represent.
+Note that subsequent uses of a node don't take its setup information
+again, so, if we wanted to debug `button` output to console,
+we would do:
+
+```
+button OUT -> IN _(console)
```

Comment:
document `_` is about anonymous nodes. Or give it a name.

I'd document that and also explain a bit about console properties, like prefix/suffix options and that the prefix defaults to the node name, in tha case of it being anonymous it will be auto-generated based on line and column.